### ISSUE TITLE
Feature: Add reduce_average (weighted reduce_mean)

### ISSUE THREAD

**Issue by cancan101** [tensorflow/tensorflow#7422]
[Numpy has a function `average`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.average.html) which peforms a weighted mean. I suggest adding this function to tensorflow or adding an optional `weights` argument to `reduce_mean`.

**Comment by kingtaurus** [comment]
Could you just use broadcasting and call `reduced_mean`?

```
tf.reduced_mean(weights * x)
```

**Comment by cancan101** [comment]
Yes though also want to divide through by sum of weights since it may not sum to 1

**Comment by girving** [comment]
@aselle Thoughts?  It seems pretty trivial to implement (`tf.reduce_sum(weights * x) / tf.reduce_sum(weights)`) so I'm not sure we need it, but it does exist in numpy.

**Comment by aselle** [comment]
As we've gotten to 1.0, there is now more careful review of API additions. I'm also working on finishing up a proposal on sub-modules and aliases. My feeling is that we should probably minimize what we add to the root tf namespace (even NumPy compatible aliases/macro functionality) and defer those to the planned numpy submodule. The idea would be that you get more numpy compatible names in the numpy sub-module. That proposal isn't quite ready yet, but that's where I would imagine average() going.

**Comment by jostmey** [comment]
+1

I hand code a weighted average all the time in my models! I would love a built in function

**Comment by AshwiniReddyChalla** [comment]
+1
I have to explicitly handle this. An optional weight parameter to reduce_mean like in mean_squared_error would be great.

**Comment by s4mpl3d** [comment]
What would the usecase of this parameter?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by aselle** [comment]
Seems reasonable, but I won't have time to do it now, so marking it contributions welcome.

**Comment by KonstZ** [comment]
This seems to work faster
def reduce_average(x, weight):
	return tf.squeeze(tf.matmul(x, weight, transpose_a=True), axis=-1)

**Comment by oxinabox** [comment]
Can this now be closed?
I see docs for
https://www.tensorflow.org/api_docs/python/tf/nn/weighted_moments

**Comment by mohantym** [comment]
Hi @cancan101 ! Does [weighted_moments](https://www.tensorflow.org/api_docs/python/tf/nn/weighted_moments) api address this feature request?
